sympy/core/mul.py

```diff
@@ -486,12 +486,15 @@ def _gather(c_powers):
         num_rat = []
         for e, b in comb_e.items():
             b = cls(*b)
-            if e.q == 1:
-                coeff *= Pow(b, e)
+            c = Pow(b, e)
+            if e.q == 1 and c.is_Number:
+                coeff *= c
```

Comment:
We want to eliminate these calls to `Pow(b, e)` but this now calls `Pow(b, e)` unconditionally.